NAME: "Margot Wolsink 2009813"

# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [8]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [9]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5         V6        V7       
1 0.7163122 0.8888078 0.8909634 0.7542971 0.08509548 0.3661351 0.1484671
2 0.7270679 0.4913349 0.8987208 0.9785760 0.95401893 0.2152560 0.8833512
  V8        V9        V10       V11       V12       V13       V14      
1 0.2006054 0.6108913 0.8869581 0.4522104 0.1943711 0.2829043 0.1523665
2 0.9456955 0.8023383 0.3468998 0.9877178 0.6460076 0.3256144 0.7943943
  V15       V16       V17        V18       V19       V20      
1 0.4702332 0.1999238 0.57400613 0.2395873 0.4685716 0.1599718
2 0.2358748 0.0715253 0.02274278 0.1996763 0.5454223 0.5165873

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [10]:
df2 <- df1 %>%
  gather(Key,Value,V1:V20)
  

head(df2)

Key Value    
1 V1  0.7163122
2 V1  0.7270679
3 V2  0.8888078
4 V2  0.4913349
5 V3  0.8909634
6 V3  0.8987208

## Assignment 2

We will generate some data:

In [11]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [12]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [13]:
df5 <- full_join(df3,df4,"id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook